test/integration-serial/serial_task_run_test.rb

```diff
@@ -28,4 +28,57 @@ def test_run_without_verify_result_fails_if_pod_was_not_created
       "Kubeclient::HttpError: Pod with same name exists"
     ], in_order: true)
   end
+
+  # We can't ensure that all the metrics are specific test when they are running in parallel
+  # so we run them serially
+  def test_failure_statsd_metric_emitted
+    deploy_task_template
+    task_runner = build_task_runner
+
+    result = false
+    metrics = capture_statsd_calls do
+      result = task_runner.run(run_params.merge(args: ["/not/a/command"]))
+    end
+
+    assert_task_run_failure(result)
+
+    metric_tags = metrics.detect { |m| m.tags.include? "namespace:#{@namespace}" }.tags
```

Comment:
I noticed in your previous version that when this fails, it fails with a no method error because of `nil.tags`. Can we change it to fail with a nice "No metrics found for this test" message?